### DISCUSSION TITLE
McSim part of Sasview

### DISCUSSION THREAD

**Discussion by andreashlarsen** [SasView/sasview#2201] [Ideas]
Following up on discussion at the development meeting on the 15th Feb 2022. 

I suggest to add McSim as a Tool in Sasview

McSim can
(1) generate point densities in varying shapes and with varying contrasts, from user inputs
(2) calculate the p(r) and I(q) from theses shapes. Optional simple structure factor and polydispersity can be added by user

McSim can be used via the GenApp GUI: 
https://somo.chem.utk.edu/mcsim/

The source code is here: 
https://github.com/ehb54/GenApp-McSim

GUI is json, rest is python3

The idea is that McSim should be used as a good tool for teaching. E.g.: a scattering curve can be generated in McSim and subsequently fitted with one of the analytical form factors in SasView. This could be used in a tutorial as part of a small-angle scattering course. 

long-term, it could be interesting to use McSim or a similar tool for fitting as well, which would be interesting and useful, but that would need a lot of work. 

note: this is also mentions in issue #2011

**Comment by lucas-wilkins** [comment]
As there seems to be nobody volunteering to take this on, I'm moving this issue to a discussion.

**Comment by andreashlarsen** [comment]
update: McSim has changed name to Shape2SAS: 
https://somo.chem.utk.edu/shape2sas/
we are working on a sasview plugin
